src/System.Security.Cryptography.Algorithms/tests/RandomNumberGeneratorTests.netcoreapp.cs

```diff
@@ -92,5 +94,132 @@ public static void Fill_RandomDistribution()
+
+        [Fact]
+        public static void GetInt32_DoesNotProduceSameNumbers()
+        {
+            int result1 = RandomNumberGenerator.GetInt32(int.MinValue, int.MaxValue);
+            int result2 = RandomNumberGenerator.GetInt32(int.MinValue, int.MaxValue);
+            int result3 = RandomNumberGenerator.GetInt32(int.MinValue, int.MaxValue);
+
+            // The changes of this happening are (2^32 - 1) * 3.
+            Assert.False(result1 == result2 && result2 == result3, "Generated the same number three times in a row.");
+        }
+
+        [Fact]
+        public static void GetInt32_FullRange_DistributesBitsEvenly()
+        {
+            // This test should work since we are selecting random numbers that are a
+            // Power of two minus one so no bit should favored.
+            int numberToGenerate = 256;
+            byte[] bytes = new byte[numberToGenerate * 4];
+            Span<byte> bytesSpan = bytes.AsSpan();
+            for (int i = 0, j = 0; i < numberToGenerate; i++, j += 4)
+            {
+                int result = RandomNumberGenerator.GetInt32(int.MinValue, int.MaxValue);
+                Span<byte> slice = bytesSpan.Slice(j, 4);
+                BinaryPrimitives.WriteInt32LittleEndian(slice, result);
+            }
+            RandomDataGenerator.VerifyRandomDistribution(bytes);
+        }
+
+        [Fact]
+        public static void GetInt32_CoinFlipLowByte()
+        {
+            int numberToGenerate = 1024;
+            Span<int> generated = stackalloc int[numberToGenerate];
+
+            for (int i = 0; i < numberToGenerate; i++)
+            {
+                generated[i] = RandomNumberGenerator.GetInt32(0, 2);
+            }
+            VerifyDistribution(generated, 0.5);
+        }
+
+
+        [Fact]
+        public static void GetInt32_CoinFlipOverByteBoundary()
+        {
+            int numberToGenerate = 1024;
+            Span<int> generated = stackalloc int[numberToGenerate];
+
+            for (int i = 0; i < numberToGenerate; i++)
+            {
+                generated[i] = RandomNumberGenerator.GetInt32(255, 257);
+            }
+            VerifyDistribution(generated, 0.5);
+        }
+
+        private static void VerifyDistribution(ReadOnlySpan<int> numbers, double expected)
+        {
+            var observedNumbers = new Dictionary<int, int>(numbers.Length);
+            for (int i = 0; i < numbers.Length; i++)
+            {
+                int number = numbers[i];
+                if (!observedNumbers.TryAdd(number, 1))
+                {
+                    observedNumbers[number]++;
+                }
+            }
+            const double tollerance = 0.07;
+            foreach ((_, int occurences) in observedNumbers)
+            {
+                double percentage = occurences / (double)numbers.Length;
+                Assert.True(Math.Abs(expected - percentage) < tollerance, "Occured number of times within threshold.");
+            }
+        }
     }
```

Comment:
It would be good to see range validity test for negative bounds, and some tests with interesting-looking bounds sizes (really I want to test both the lzcnt implementation and the algorithm.  Something annoying like (x, x + 0x0101) so the high bit in the mask value is technically required, but rarely valid.  If you ran that 1000 times you "should" have at least one hit on the highest value, and shouldn't have value skew.)

```
(int.MinValue, int.MinValue + 3)
(-257, -129)
(-100, 5)
(254, 512)
...
```

Also, roll 1000d6, check for skew.  (The existing "coinflip" test sort of does this, but it's bit-aligned in range, so it doesn't test the high value rejection vs mod bias algorithm differential)